### PR TITLE
Initial clean up and preparation for Ruby 2.2+ only support

### PR DESCRIPTION
**Pull request by deepj**
I've started working on some small improvements. I mostly remove some dead code and some obsolete things such as Mongrel. A question here is if Rack is going to support things like Memcached sessions (based on legacy library), lighttp and so on. Any suggestions?

### CODE REVIEW

**Comment by tenderlove** [comment]
@deepj I'm really sorry. I didn't see this PR. Can you rebase it and I'll apply the rest of your changes? Thank you so much for this work!

**Comment by deepj** [comment]
@tenderlove done

**Comment by tenderlove** [comment]
Thanks!

**Comment by spastorino** [comment]
:+1: I was going to merge this one thanks t.lo

File: lib/rack/utils.rb [hunk 1]
```diff
@@ -360,19 +360,6 @@ def delete_cookie_header!(header, key, value = {})
...
-      def bytesize(string)
-        string.bytesize
-      end
-    else
-      def bytesize(string)
-        string.size
-      end
-    end
-    module_function :bytesize
-
```
Comment (jeremy) [inline lib/rack/utils.rb]: For the sake of gracefully deprecating breaking changes to public API, suggest keeping this as
```ruby
def bytesize(string)
  warn 'Rack::Utils.bytesize is deprecated in Rack 2.0 and will be removed in Rack 2.1. Use String#bytesize instead.'
  string.bytesize
end
```

Ref. https://github.com/rack/rack-contrib/pull/124 which used this public API.

Comment (glaucocustodio) [inline lib/rack/utils.rb]: Please take a look at https://github.com/rack/rack/pull/1103 cc/ @jeremy

Comment (jeremy) [inline lib/rack/utils.rb]: Thanks @glaucocustodio.

**Pull request merged**